app/src/ui/changes/changes-list.tsx

```diff
@@ -168,6 +179,10 @@ export class ChangesList extends React.Component<IChangesListProps, {}> {
     const filesDescription = `${fileCount} changed ${filesPlural}`
     const anyFilesSelected =
       fileCount > 0 && this.includeAllValue !== CheckboxValue.Off
+    const invalidationProps = {
```

Comment:
This will invalidate the list on every single render since you're creating a new object each time. react-virtualized behaves as a PureComponent which means shallow comparisons. You'll have to compute this invalidation object when, and only if, these props changes and store the invalidation object in state which you can then pass to the list in render.

Alternatively we could refactor this component to pass an `onContextMenu` handler down to each list item and handle the context menu in this class instead.